utils/namespaces.js

```diff
@@ -0,0 +1,33 @@
+import { sortBy } from '@/utils/sort';
+
+// This method is used by the NameNsDescription component and by the create/edit
+// workload form to provide namespaces to choose from.
+const getAvailableNamespaces = (resource) => {
+  const inStore = resource.$store.getters['currentStore'](resource.namespaceType || 'namespace');
+  const choices = resource.$store.getters[`${ inStore }/all`](resource.namespaceType || 'namespace');
+
+  const out = sortBy(
+    choices.filter( (N) => {
+      const isSettingSystemNamespace = resource.$store.getters['systemNamespaces'].includes(N.metadata.name);
+
+      return resource.isVirtualCluster ? !N.isSystem && !N.isFleetManaged && !isSettingSystemNamespace : true;
```

Comment:
It is not clear why all these 3 values must be `true`. Wrapping the conditions in a variable may probably give an understandable purpose. E.g. `hasRequirements` or `hasVirtualClusterDependencies`; please bare mind that I have no idea what's behind the logic, so the names may be not appropriate.

Reply:
I moved this code out of the NameNsDescription component so I could reuse it in another component. Although I didn't write it, that's a very good question. I have more questions about that component in general - I'll see what I can find out.